internal/provider/bms/bms_types.go

```diff
@@ -0,0 +1,139 @@
+package bms
+
+import (
+	"context"
+
+	timeoutsD "github.com/hashicorp/terraform-plugin-framework-timeouts/datasource/timeouts"
+
+	supertypes "github.com/FrangipaneTeam/terraform-plugin-framework-supertypes"
+
+	v1 "github.com/orange-cloudavenue/cloudavenue-sdk-go/v1"
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/pkg/utils"
+)
+
+type (
+	bmsModelDatasource struct {
+		ID       supertypes.StringValue                                   `tfsdk:"id"`
+		Timeouts timeoutsD.Value                                          `tfsdk:"timeouts"`
+		Env      supertypes.SetNestedObjectValueOf[bmsModelDatasourceEnv] `tfsdk:"env"`
+	}
+
+	bmsModelDatasourceEnv struct {
+		Network supertypes.SetNestedObjectValueOf[bmsModelDatasourceNetwork] `tfsdk:"network"`
+		BMS     supertypes.SetNestedObjectValueOf[bmsModelDatasourceBMS]     `tfsdk:"bms"`
+	}
+
+	bmsModelDatasourceNetwork struct {
+		VLANID supertypes.StringValue `tfsdk:"vlan_id"`
+		Subnet supertypes.StringValue `tfsdk:"subnet"`
+		Prefix supertypes.StringValue `tfsdk:"prefix"`
+	}
+
+	bmsModelDatasourceBMS struct {
+		Hostname          supertypes.StringValue                                             `tfsdk:"hostname"`
+		BMSType           supertypes.StringValue                                             `tfsdk:"bms_type"`
+		OS                supertypes.StringValue                                             `tfsdk:"os"`
+		BiosConfiguration supertypes.StringValue                                             `tfsdk:"bios_configuration"`
+		Storage           supertypes.SingleNestedObjectValueOf[bmsModelDatasourceBMSStorage] `tfsdk:"storage"`
+	}
+
+	bmsModelDatasourceBMSStorage struct {
+		Local  supertypes.ListNestedObjectValueOf[bmsModelDatasourceBMSStorageGen] `tfsdk:"local"`
+		System supertypes.ListNestedObjectValueOf[bmsModelDatasourceBMSStorageGen] `tfsdk:"system"`
+		Data   supertypes.ListNestedObjectValueOf[bmsModelDatasourceBMSStorageGen] `tfsdk:"data"`
+		Shared supertypes.ListNestedObjectValueOf[bmsModelDatasourceBMSStorageGen] `tfsdk:"shared"`
+	}
+
+	bmsModelDatasourceBMSStorageGen struct {
```

Comment:
```suggestion
	bmsModelDatasourceBMSStorageDetail struct {
```

Reply:
Gen for generic, but detail is the good name use in sdk too: Fix